src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -287,7 +291,8 @@ void ZeroInferRequest::set_tensor(const ov::Output<const ov::Node>& port, const
             _logger.debug("ZeroInferRequest::set_tensor - got the same tensor, do nothing");
             return;
         }
-        if (is_batched_input(foundPort.idx)) {
+        std::vector<ov::SoPtr<ov::ITensor>> tensorVector = {tensor};
+        if (is_batched_input(foundPort.idx) || _graph->get_batch_size(_metadata, tensorVector).has_value()) {
```

Comment:
This code in set_tensor() belongs to the use-case "batch in compiler" it seems we don;t have to intrude here, as dynamic case in compile for batch is not supported yet